.github/workflows/node.js.yml

```diff
@@ -0,0 +1,59 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build
+      - run: npm test
+      - name: Archive build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: build-${{ matrix.node-version }}
```

Comment:
Do we surely need all these artefacts for all version of node.js ?

When we look into build process: `"build": "tsc -p tsconfig.json"`, there is only typescript compiling which is not depend on Node.js version and result is always the same (so is there some sense to build and upload the same result code).

Reply:
Sure, can upload only artifact which will be released.